### PR TITLE
Implement token refresh logic

### PR DESCRIPTION
**Pull request by jviotti**

### CODE REVIEW

File: lib/request.coffee [hunk 1]
```diff
@@ -45,12 +46,26 @@ prepareOptions = (options = {}) ->
...
+	Promise.try ->
+		return if not options.refreshToken
+
+		utils.shouldUpdateToken().then (shouldUpdateToken) ->
+			return if not shouldUpdateToken
+
+			exports.send
+				url: '/whoami'
+				refreshToken: false
+			.get('body').then (newToken) ->
```
Comment (Page-) [inline lib/request.coffee]: `.then(token.set)`?

File: lib/request.coffee [hunk 2]
```diff
@@ -45,12 +46,26 @@ prepareOptions = (options = {}) ->
...
+				token.set(newToken)
+
+	.then ->
```
Comment (Page-) [inline lib/request.coffee]: I personally prefer
```
.then(utils.getAuthorizationHeader)
.then (authorizationHeader) ->
```
or

```
.then ->
	utils.getAuthorizationHeader()
.then (authorizationHeader) ->
```
as it avoids the need for the additional indentation

Comment (jviotti) [inline lib/request.coffee]: Missed that one. Fixed!

File: lib/request.coffee (hunk 1 above)
Comment (jviotti) [inline lib/request.coffee]: True. I moved to another line to add a `console.log` for debugging purposes and forgot to collapse it back.

**Comment by Page-** [comment]
LGTM

**Pull request merged**